lib/services/plugin_static_credentials_test.go

```diff
@@ -45,4 +46,24 @@ func TestMarshalPluginStaticCredentialsRoundTrip(t *testing.T) {
 	unmarshaled, err := UnmarshalPluginStaticCredentials(payload)
 	require.NoError(t, err)
 	require.Empty(t, cmp.Diff(creds, unmarshaled))
+
+	// TODO(greedy52) PluginStaticCredentials uses protojson for marshaling
+	// and unmarshaling. Unfortunately Expires (*time.Time) is not supported
+	// and becomes zero after the marshaling. Unlikely Expires will ever be
+	// needed for PluginStaticCredentials in production but this can cause
+	// unexpected issues in other places (like resource.SetExpiry in cache
+	// tests).
```

Comment:
Yikes. We would probably need to move to V2 resource to fix this issue?

Reply:
Yeah, I was trying some hacks, like marshaling header/spec separately, but I couldn't make it backwards compatible. We may have to do an upgrade to do it properly. It's not a concern for Production thought as these resources are designed to long-lived (as opposed to having a ttl)